src/OidcKeyRegistry.sol

```diff
@@ -79,4 +68,55 @@ contract OidcKeyRegistry is Initializable, OwnableUpgradeable {
   function getKeys(bytes32 issHash) public view returns (Key[MAX_KEYS] memory) {
     return OIDCKeys[issHash];
   }
+
+  function deleteKey(bytes32 issHash, bytes32 kid) public onlyOwner {
+    _deleteKey(issHash, kid);
+    _compactKeys(issHash);
+    emit KeyDeleted(issHash, kid);
+  }
+
+  function _compactKeys(bytes32 issHash) private {
+    Key[MAX_KEYS] memory keys;
+    uint8 keyCount = 0;
+    uint8 currentIndex = keyIndexes[issHash];
+
+    for (uint8 i = 0; i < MAX_KEYS; i++) {
+      uint8 circularIndex = (currentIndex + i) % MAX_KEYS;
+      if (OIDCKeys[issHash][circularIndex].kid != 0) {
+        keys[keyCount] = OIDCKeys[issHash][circularIndex];
+        keyCount++;
+      }
+    }
+
+    for (uint8 i = 0; i < keyCount; i++) {
+      OIDCKeys[issHash][i] = keys[i];
+    }
+
+    for (uint8 i = keyCount; i < MAX_KEYS; i++) {
+      delete OIDCKeys[issHash][i];
+    }
+
+    keyIndexes[issHash] = keyCount == 0 ? MAX_KEYS - 1 : (keyCount - 1) % MAX_KEYS;
```

Comment:
I imagine this is to avoid overflow, right?

What do you think about something like
```suggestion
    // Adding MAX_KEYS to take adventage of modulus and avoid overflow
    keyIndexes[issHash] = (keyCount + MAX_KEYS - 1) % MAX_KEYS;
```

In my head that's a little bit easier to parse with the eyes. Wdyt?

In my head th